sql/_08_javasp/answers/case_index_scan_01.answer

```diff
@@ -0,0 +1,166 @@
+===================================================
+in    
+1     
+
+Query plan:
+temp(distinct)
+    subplan: sscan
+                 class: dual node[?]
+                 cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+(select distinct inttest(?) from dual dual)
+Query plan:
+idx-join (inner join)
+    outer: sscan
+               class: av? node[?]
+               cost:  ? card ?
+    inner: iscan
+               class: tbl node[?]
+               index: i_tbl term[?] (covers)
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select count(*) from tbl tbl, (select distinct inttest(?) from dual dual) av? (av_?) where tbl.ord=av?.av_?
+===================================================
+in    
+10     
+
+Query plan:
+sscan
+    class: t_? node[?]
+    cost:  ? card ?
+Query stmt:
+(select tbl.ord from table({?}) t_? (c_?))
+Query plan:
+nl-join (cselect join)
+    edge:  table(?) -> t_? node[?]
+    outer: sscan
+               class: t_? node[?]
+               cost:  ? card ?
+    inner: sscan
+               class: t_? node[?]
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+(select t_?.c_?, t_?.c_? from (select tbl.ord from table({?}) t_? (c_?)) t_? (c_?), (inttest(t_?.c_?)) t_? (c_?))
+Query plan:
+temp(distinct)
+    subplan: nl-join (inner join)
+                 edge:  table(?) -> t_? node[?]
+                 outer: sscan
+                            class: tbl node[?]
+                            cost:  ? card ?
+                 inner: sscan
+                            class: t_? node[?]
+                            sargs: table(?) -> t_? node[?]
+                            cost:  ? card ?
+                 subqs: ?
+                 cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+(select distinct t_?.c_? from tbl tbl, (select t_?.c_?, t_?.c_? from (select tbl.ord from table({?}) t_? (c_?)) t_? (c_?), (inttest(t_?.c_?)) t_? (c_?)) t_? (c_?, c_?))
+Query plan:
+idx-join (inner join)
+    outer: sscan
+               class: av? node[?]
+               cost:  ? card ?
+    inner: iscan
+               class: tbl node[?]
+               index: i_tbl term[?] (covers)
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select count(*) from tbl tbl, (select distinct t_?.c_? from tbl tbl, (select t_?.c_?, t_?.c_? from (select tbl.ord from table({?}) t_? (c_?)) t_? (c_?), (inttest(t_?.c_?)) t_? (c_?)) t_? (c_?, c_?)) av? (av_?) where tbl.ord=av?.av_?
+===================================================
+like    
+1     
+
+Query plan:
+iscan
+    class: tbl node[?]
+    index: i_tbl_char term[?] (covers)
+    cost:  ? card ?
+Query stmt:
+select count(*) from tbl tbl where tbl.col_char= ?:? 
+===================================================
+like    
+1     
+
+Query plan:
+sscan
+    class: dual node[?]
+    cost:  ? card ?
+Query stmt:
+(select stringTest('a') from dual dual)
+Query plan:
+sscan
+    class: tbl node[?]
+    sargs: term[?]
+    cost:  ? card ?
```

Comment:
Currently can not index-scan the 'JAVASP' function on 'LIKE' clause.
Related http://jira.cubrid.org/browse/CBRD-24686